File: src/api/product.ts
import api from "./api";

export interface ProductRequest {
    name: string;
    sku: string;
    category: string;
    description: string;
    purchase_price: number;
    selling_price: number;
    stock_quantity: number;
    reorder_threshold: number;
}

export const getProducts = async () => {
    const response = await api.get("/products/");
    return response.data;
};

export const getProduct = (id: string) =>
    api.get(`/products/${id}`);

export const createProduct = (data: any) =>
    api.post("/products/", data);

export const updateProduct = (
    id: string,
    data: Partial<ProductRequest>
) =>
    api.put(`/products/${id}`, data);

export const deleteProduct = (id: string) =>
    api.delete(`/products/${id}`);

File: backend/app/schemas/product.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ProductBase(BaseModel):
    name: str
    sku: str
    category: str
    description: str | None = None
    purchase_price: float
    selling_price: float
    stock_quantity: int
    reorder_threshold: int


class ProductCreate(ProductBase):
    pass


class ProductUpdate(ProductBase):
    pass


class ProductResponse(ProductBase):
    id: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)